NTBS-438: Display empty space on Overview page when value doesn't exist

ffarmanov opened the pull request:
<!--- Provide a general summary of your changes in the Title above, including the related Jira ticket -->

Display empty space on Overview page when value doesn't exist

## Description
<!--- High level changes overview -->
Overview page was omitting empty values and therefore shifting some values. Fixed the bug by displaying empty space.
Also overview page was not displaying Total when one of the values is missing

## Checklist:
- [x] Automated tests are passing locally.
### Accessibility testing
- [x] Test functionality without javascript
- [x] Test in small window (immitating small screen)
- [x] Check the feature looks and works correctly in IE11
- [x] Zoom page to 400% - content still visible?
- [x] Test feature works with keyboard only operation
- [x] Test with one screen reader - logical flow, no unexpected content
- [x] Passes automated checker (e.g. [WAVE](https://chrome.google.com/webstore/detail/wave-evaluation-tool/jbbplnpkjmmeebjpijfedlgcdilocofh))

ffarmanov commented on ntbs-service/Models/Notification.cs:
```diff
@@ -94,7 +94,11 @@ private string GetNotificationStatusString()
 
         private int? CalculateSum(int? x, int? y) {
-            return x + y;
+            if (x == null && y == null) 
```
If one of the values is null we would still like to calculate total. Right @ChristianMcc ?
Assuming this I added `x ?? 0 + y ?? 0`

but then I needed to check if both values are not null at the same time, otherwise the total is 0 when both values are missing

ffarmanov commented on ntbs-service/Models/Validations/ValidationHelper.cs:
```diff
@@ -24,10 +24,10 @@ public static string ValidYearLaterThanBirthYear(int birthYear)
-        public const string ContactTracingChildrenLatentTB = "Must be smaller or equal to than number of children screened minus those with latent TB";
-        public const string ContactTracingAdultsActiveTB = "Must be smaller or equal to than number of adults screened minus those with active TB";
-        public const string ContactTracingChildrenActiveTB = "Must be smaller or equal to than number of children screened minus those with active TB";
+        public const string ContactTracingAdultsLatentTB = "Must be smaller or equal to than number of adults screened minus those with active TB";
```
Latent and active are depend on each other therefore the messages should be the other way around

jasiekmiko commented on ntbs-service/Models/Validations/ValidationHelper.cs:
```diff
@@ -24,10 +24,10 @@ public static string ValidYearLaterThanBirthYear(int birthYear)
+        public const string ContactTracingAdultsLatentTB = "Must be smaller or equal to than number of adults screened minus those with active TB";
+        public const string ContactTracingChildrenLatentTB = "Must be smaller or equal to than number of children screened minus those with active TB";
+        public const string ContactTracingAdultsActiveTB = "Must be smaller or equal to than number of adults screened minus those with latent TB";
+        public const string ContactTracingChildrenActiveTB = "Must be smaller or equal to than number of children screened minus those with laten TB";
```
typo :)

jasiekmiko commented on ntbs-service/Pages/Notifications/OverviewPartials/_ContactTracingOverviewPartial.cshtml:
```diff
@@ -17,9 +17,9 @@
-                    <div class="contact-tracing-value"> @Model.Notification.ContactTracing.AdultsIdentified </div>
-                    <div class="contact-tracing-value"> @Model.Notification.ContactTracing.ChildrenIdentified </div>
-                    <div class="contact-tracing-value"> @Model.Notification.TotalContactsIdentified </div>
+                    <div class="contact-tracing-value"> @(Model.Notification.ContactTracing.AdultsIdentified ?? Html.Raw(emptySpace))</div>
```
are we not better off by putting a minimal width on the `contact-tracing-value` class?

jasiekmiko replied:
Wait, that's not right - our markup is the wrong way round! I think using a table here would be the cleanest approach actually..

ffarmanov commented:
@jasiekmiko I have fixed the issue for Total values displaying 0 when landing on a page. It's raised as a separate bug, but I have added it to this branch with appropriate commit message

lbbethke commented on ntbs-service/Models/ContactTracing.cs:
```diff
@@ -43,5 +44,22 @@ public class ContactTracing : ModelBase
 
         [PositiveIntegerSmallerThanValue("ChildrenStartedTreatment", ErrorMessage = ValidationMessages.ContactTracingChildrenFinishedTreatment)]
         public int? ChildrenFinishedTreatment { get; set; }
+
```
@jasiekmiko I remember talking about defining these on the individual models rather than Notification at some other point, but there was a reason not to do it... what has changed since?

ffarmanov replied:
This were a few reasons
- It's nicer to understand which models owns this fields
- Total was not displayed when landing on Contact Tracing page. So, I just needed to display Total instead of 0 at the beginning.
- View is a bit cleaner (less chaining when accessing parameters)

jasiekmiko replied:
So I still think having them on `Notification` is nicer, as I think the consumer of these *doesn't* want to care about which sub-model they're coming from - they're all part of the notification record. I would argue having them on the `Notification` is how we get less chaining on the overview page (`Model.Total...` vs `Model.ContactTracing.Total...`).

But, that logic breaks apart once we start using these values on the individual pages, and it looks like we're using the `totals` this way here, as the last thing we'd want is to have them on both.

This leaves us in a bit of an unfortunate state, where we have most of them on Notification but some on the individual record, which could get confusing. Perhaps something to keep an eye on and see if we end up with more cases like this?
